src/pymovements/utils/parsing.py

```diff
@@ -194,6 +214,24 @@ def parse_eyelink(
             for additional_column in additional_columns:
                 samples[additional_column].append(current_additional[additional_column])
 
+        if compiled_metadata_patterns:
+            for pattern in compiled_metadata_patterns:
+                if match := pattern.match(line):
+                    for column, value in match.groupdict().items():
+                        metadata[column] = value
+
+                    # each metadata pattern should only match once
+                    compiled_metadata_patterns.remove(pattern)
+
+    # note sure if this is the best way...
```

Comment:
I'd propose to do it this way:

```python
if not metadata:
    raise Warning()

metadata['version_number'], metadata['model'] = _parse_full_version(
    metadata['version_1'], metadata['version_2'],
)
```

I originally thought that we would also need to check if `version_1` and `version_2` is in the parsed metadata, but as it's a `defaultdict` you solved it elegantly already.

But anyway, please leave a comment there that the key/values will be generated on the fly (empty string) if the keys are not available.

Reply:
Ah now I see, that there's difference in output between the two:
My proposal would create `version_number` and `model` regardless if metadata has been parsed.

What do you think, which way would make most sense to you? I think both are valid options.

But anyway, please leave a comment there that the key/values will be generated on the fly (empty string) if the keys are not available.

Reply:
I left it like it was for now, but we can still adapt it.